admin/class-boldgrid-backup-admin-auto-updates.php

```diff
@@ -86,24 +103,64 @@ public function __construct() {
 	 * @param string $option Name of option being updated.
 	 */
 	public function wordpress_option_updated( $old_value, $new_value, $option ) {
-		// Determines if this is being fired for a theme or plugin update
-		$update_type = 'auto_update_plugins' === $option ? 'plugins' : 'themes';
-		$core        = apply_filters( 'boldgrid_backup_get_core', null );
-		$settings    = $core->settings->get_settings();
+		// Determines if this is being fired for a theme or plugin update.
+		$this->auto_update_log = new Boldgrid_Backup_Admin_Log( $this->core );
+		$this->auto_update_log->init( 'auto-update.log' );
+		$update_type    = 'auto_update_plugins' === $option ? 'plugins' : 'themes';
+		$this->settings = $this->core->settings->get_settings();
+		$wpcs           = isset( $this->settings['auto_update']['wpcore'] ) ? $this->settings['auto_update']['wpcore'] : [];
+
+		// Log the Core auto update settings only once.
+		if ( false === $this->core_save_settings ) {
+			$this->auto_update_log->add( '--------------------------------------------------' );
+			$this->auto_update_log->add( 'WP Core auto update settings' );
+
+			if ( isset( $wpcs['all'] ) && $wpcs['all'] ) {
+				$this->auto_update_log->add( 'WP Core Updates are set to auto update all.' );
+			}
+			$dev = ( ! empty( $wpcs ) && $wpcs['dev'] ) ? 'true' : 'false';
+			$this->auto_update_log->add( 'Setting WP Core Dev Updates to ' . $dev . '.' );
+			$major = ( ! empty( $wpcs ) && $wpcs['major'] ) ? 'true' : 'false';
+			$this->auto_update_log->add( 'Setting WP Core Major Updates to ' . $major . '.' );
+			$minor = ( ! empty( $wpcs ) && $wpcs['minor'] ) ? 'true' : 'false';
+			$this->auto_update_log->add( 'Setting WP Core Minor Updates to ' . $minor . '.' );
+			$translation = ( ! empty( $wpcs ) && $wpcs['translation'] ) ? 'true' : 'false';
+			$this->auto_update_log->add( 'Setting WP Core Translation Updates to ' . $translation . '.' );
+			$this->auto_update_log->add( '--------------------------------------------------' );
+
+			$this->core_save_settings = true;
+		}
 
 		// The plugins / themes listed in $new_value will only be those that have auto updates enabled.
 		$enabled_offers = $new_value;
-		foreach ( $settings['auto_update'][ $update_type ] as $offer => $enabled ) {
+
+		$this->auto_update_log->add( 'WP ' . $update_type . ' auto update settings.' );
+
+		foreach ( $this->settings['auto_update'][ $update_type ] as $offer => $enabled ) {
 			// Do not modify the 'default' setting. This is used to define the default for new themes / plugins.
 			if ( 'default' === $offer ) {
 				continue;
 			}
+
+			if ( 'plugins' === $update_type ) {
+				$item_data = get_plugin_data( WP_PLUGIN_DIR . '/' . $offer );
+				$item_name = $item_data['Name'];
+			} else {
+				$item_data = get_plugin_data( get_template_directory() . '/../' . $offer . '/style.css' );
```

Comment:
$item_data doesn't seem to be used here.... Instead you're setting `$item_name = $offer` 
This line could probably go away.... Especially since I don't think it's actually doing anything, since you would use wp_get_theme() for this

Reply:
this is updated